# Using apps to attract users

This project is intended to show what type of apps most users prefer.  The goal is to create a frequency table of the most downloaded apps by their type in order to attract users to more profitable apps.  Some things to be noted:

1. The dataset presented here is for the _Apple App Store_ and _Google Play Store_ only, since they are the most   widely used and cover 75% of the apps market.  

2. A sample size was used for each market:
_Google Play_ sample size [dataset](https://www.kaggle.com/lava18/google-play-store-apps/downloads/google-play-store-apps.zip/6) uses 10,000 Android apps and _Apple App_ sample size [dataset](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home) uses 7,000 iOS apps.

In [48]:
#Applestore
opened_file = open("AppleStore.csv")
from csv import reader
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

#Google Store
opened_file = open("googleplaystore.csv")
from csv import reader
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]


# Initial analysis of the datasets

In our intial, exploratory analysis of the two data sets, we will define a function named explore_data().  This will allow us to view multiple rows in a more readable way.  It will end with an option to view the total number of rows and columns for that data set.

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')#adds a new empty line
    if rows_and_columns:
        print('Number of rows: ', len(dataset))
        print('Number of columns: ', len(dataset[0]))

In [3]:
#Apple store data
print(ios_header)
print('\n')
explore_data(ios,3,6, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']


['5', '282935706', 'Bible', '92774400', 'USD', '0', '985920', '5320', '4.5', '5', '7.5.1', '4+', 'Reference', '37', '5', '45', '1']


['6', '283619399', 'Shanghai Mahjong', '10485713', 'USD', '0.99', '8253', '5516', '4', '4', '1.8', '4+', 'Games', '47', '5', '1', '1']


Number of rows:  7197
Number of columns:  17


We see that the Apple Store data set has 7,197 apps and 17 columns.  At a quick glance, the columns that might be useful for our analysis are: `track_name`, `price`, `currency`, `prime_genre`, `rating_count_tot` (rating count total for all versions) and `rating_count_ver` (the rating count for the current version).  More information about the column names can be found [here](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps/home).

In [4]:
#Google Play data
print(android_header)
print('\n')
explore_data(android,3,6,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows:  10841
Number of columns:  13


We can see that the Google Play dataset has 10,841 rows and 13 columns. The columns that will probably be most useful for our analysis will be `App`, `Category`, `Reviews`, `Installs`, `Type`, `Price` and `Genres`.

# Cleanup & Deletion

The Google Play dataset has a dedicated [discussion section](https://www.kaggle.com/lava18/google-play-store-apps/discussion).  We can see by one of the [discussion topics](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015), that there is an outlier error for row 10472.  Let's print this row and compare it against the header and another row that is correct.


In [5]:
print (android_header)
print('\n')
explore_data(android,10471,10473)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']




We can see that the row of "Life Made Wi-Fi Touchstone Photo Frame" is missing it's `category`.  This causes rating to be viewed in the category field and shifts all subsequent values into the wrong columns.  Since we cannot infer a category from the information given, we will delete this record.

In [15]:
#print(len(android))
#del android[10472] #only to be run ONCE! 
#print(len(android))

10841
10840


# Cleaning & Duplicates

There is [discussion](https://www.kaggle.com/lava18/google-play-store-apps/discussion/67894) in the Google Play data set as to multiple entries for the same app.  We will create a list of duplicates to explore what has happened.

In [16]:
duplicate_apps =[]
unique_apps=[]

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)

print('Number of duplicate apps: ', len(duplicate_apps))
print('\n')
print('Examples of duplicate apps: ', duplicate_apps[:15])

Number of duplicate apps:  1181


Examples of duplicate apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


It seems one reason for duplicates to occur is because of review counts.  The data has been collected at different times and the review count reflects that.  The higher the number of reviews, the more recent the data is.  For this reason, our criteria is to only keep the duplicate with the highest number of reviews while removing the others.  

Therefore we will:

- Create a dictionary where each key is a unique app name, and the value is the highest number of reviews of that app.
- Use that new dictionary to create a new data set, which will only have one entry per app.

Let's start with the dictionary:

In [17]:
reviews_max={}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
        
print('Expected length: ', len(android) - len(duplicate_apps))
print('Actual length: ', len(reviews_max))

Expected length:  9659
Actual length:  9659


Now let's remove the duplicates.  Again, we only want to keep the duplicate entry with the highest number of reviews.

- We will create two empty lists, android_clean and already_added.
- Next we loop through the android data set, and for **each iteration**:
    - We isolate the name of the app and the number of reviews
    - We add the current row to the android_clean list as well as the app name to the already_cleaned list if:
        - The number of reviews of the current app matches the number of reviews of that app as describd in the reviews_max dictionary; AND
        - The name of the app doesn't exist in the already_added list.  *(This is needed because for some apps, the highest number of reviews appears more than once as duplicate records.)*

In [18]:
android_clean=[]
already_added=[]

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (n_reviews == reviews_max[name]) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)
        

Let's explore the data and make sure we get the expected value of 9659.  Which again is expected because it is the length of the original dataset of Android(10840) minus the number of duplicates(1181).

In [19]:
explore_data(android_clean,3,6,True)

['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


['Smoke Effect Photo Maker - Smoke Editor', 'ART_AND_DESIGN', '3.8', '178', '19M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'April 26, 2018', '1.1', '4.0.3 and up']


Number of rows:  9659
Number of columns:  13


We have the 9659 as expected.

# Removing Non-English Apps

Our market is for English speaking users. On further examination of the datasets, we see that not all apps are English only apps.

In [24]:
print(ios[813][1])
print(android[6731][1])

print(android_clean[4412][0])
print(android_clean[7940][0])

436672029
FAMILY
中国語 AQリスニング
لعبة تقدر تربح DZ


We're not interested in keeping these kind of apps, so we'll remove them. One way to go about this is to remove each app whose name contains a symbol that is not commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;, etc.), and other symbols (+, *, /, etc.).

All these characters that are specific to English texts are encoded using the ASCII standard. Each ASCII character has a corresponding number between 0 and 127 associated with it, and we can take advantage of that to build a function that checks an app name and tells us whether it contains non-ASCII characters.

We built this function below, and we use the built-in ord() function to find out the corresponding encoding number of each character.

In [36]:
def eng(string):
    
    for character in string:
        if ord(character) > 127:
            return False
        
    return True
        
print(eng('Instagram'))
print(eng('爱奇艺PPS -《欢乐颂2》电视剧热播'))


True
False


Some of the apps in our list use symbols or emojis in their title, though their apps function in English.  These charcaters fall outside of the ASCII range.  Because of this, we'll remove useful apps if we use the function in its current form.

In [39]:
print(eng('Docs To Go™ Free Office Suite'))
print(eng('Instachat 😜'))

print(ord('™'))
print(ord('😜'))

False
False
8482
128540


We will remove an app if it's name contains more than three non-ASCII characters:

In [41]:
def eng(string):
    non_ascii = 0
    
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    
    if non_ascii > 3:
        return False
    else:
        return True



There may be a few non-English apps that get through the cracks, but the chances are so small that it gives us enough English apps to work with for a strong final analysis.  

In [50]:
ios_eng = []
android_cl_eng = []

for app in android_clean:
    name = app[0]
    if eng(name):
        android_cl_eng.append(app)
        
for app in ios:
    name = app[2]
    if eng(name):
        ios_eng.append(app)
        
explore_data(android_cl_eng,0,3,True)
print('\n')
explore_data(ios_eng,0,3,True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows:  9614
Number of columns:  13


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '1

We can see that we are left with 9614 Android apps and 6183 iOS apps.

# Isolating Free Apps

We only build apps that free to download and install, since our main source of revenue is in-app ads.  Therefore, our last cleaning technique is to remove all paid apps from our dataset.  We will do this by isolating only the free apps in both data sets.

In [52]:
ios_final = []
android_final = []

for app in android_cl_eng:
    price = app[7]
    if price == '0':
        android_final.append(app)

for app in ios_eng:
    price = app[5]
    if price == '0':
        ios_final.append(app)
        
print("iOS is: ", len(ios_final))
print('Android is: ', len(android_final))

iOS is:  3222
Android is:  8864


Our final dataset count for each app store is 3222 for iOS and 8864 for Android.  This is a large enough sample set for our analysis.

# Analysis: Most Common Apps by Genre

Our strategy is to build an app for the Google Store based off of which genre is most popular.  If the app is profitable within 6 months in the Google Store, we will build a version in the Apple Store.  Since the end goal is to have the app profitable in both the Google Play and Apple stores, we will want to find genres that are popular in both markets.  

Our first step is to identify which genres are most *common* for each market.  We will do this with a frequency table for the `prime_genre` column in the Apple store, as well as the `Genres` and `Category` columns in the Google Play store.

Our frequency table will come in two steps.  The first will be to define a function showing frequencies as a percentage of popularity, and the second function will display those frequencies in descending order.


In [53]:
def freq_table(dataset,index):
    table={}
    total = 0
    
    for row in dataset:
        total += 1
        value = row[index]
        if value in table:
            table[value] += 1
        else:
            table[value] = 1
            
    table_percentages={}
    for key in table:
        percentage = (table[key] / total) * 100
        table_percentages[key] = percentage
        
    return table_percentages

def display_table(dataset,index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        keyval_tuple = (table[key], key)
        table_display.append(keyval_tuple)
        
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
    

Let's start with the `prime_genre` in the iOS Apple Store:

In [54]:
display_table(ios_final,12)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


We can see that most English speaking apps are Games, which consume over half of the market at 58%.  Next are entertainment with 8%, followed by Photo/Video at 5%.  The next two genres are make up about 3% each of the market: Education and Social Networking.  

This tells us that most of the English apps created for the iOS Apple Store are for entertainment (Games, Entertainment, Photo/Video, Social Networking, Shopping, Sports, and Music).  Apps designed for practical activities like Education, utilities, Health & Fitness, Lifestyle, and News are more rare.  But, just because they occur more frequently, doesn't make them more popular to download.  The demand may be different than the offer.

Now let's examine the Google Play store at their `Genres` and `Categories`.

In [55]:
display_table(android_final,1) #Category

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

We can see a different picture emerges in the Google Play store.  There are far more practical apps designed for use on Android.  This is also confirmed under the `Genres` category:

In [56]:
display_table(android_final,9) #Genres

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

We aren't sure what the relationship is between `Category` and `Genres`, but `Genres` seems more granular than what we need for our Analysis.  Therefore, we will only work with the `Category` column within the Google Play store.

So far, our Analysis shows that more apps for fun have been designed for the iOS store, while there is a better balance of fun and productive apps designed for the Google Play store.

Now let's compare what kind of apps have the most users.

# Most Popular Apps by Genre

The method that we will use to find which apps are most popular is to calculate the average number of installs for each genre.  In the Google Play dataset, we can use the `Installs` column.  For the Apple store, we will use the `rating_count_tot`.

We will start with the **iOS Apple** store:

In [62]:
ios_genre=freq_table(ios_final, 12)

for genre in ios_genre:
    total = 0 #the sum of user ratings
    len_genre = 0 #the number of apps specific to ea genre
    for app in ios_final:
        genre_app = app[12]
        if genre_app == genre:
            n_ratings = float(app[6])
            total += n_ratings
            len_genre += 1

    ios_avg_rating = total / len_genre
    print(genre, ':', ios_avg_rating)

Productivity : 21028.410714285714
Weather : 52279.892857142855
Shopping : 26919.690476190477
Reference : 74942.11111111111
Finance : 31467.944444444445
Music : 57326.530303030304
Utilities : 18684.456790123455
Travel : 28243.8
Social Networking : 71548.34905660378
Sports : 23008.898550724636
Health & Fitness : 23298.015384615384
Games : 22788.6696905016
Food & Drink : 33333.92307692308
News : 21248.023255813954
Book : 39758.5
Photo & Video : 28441.54375
Entertainment : 14029.830708661417
Business : 7491.117647058823
Lifestyle : 16485.764705882353
Education : 7003.983050847458
Navigation : 86090.33333333333
Medical : 612.0
Catalogs : 4004.0


At first glance the Navigation, Reference, Social Networking and Music genres seem to dominate the user approval rating.  But, once you drill down into those genres, we see that they are dominated by big user apps like Google Maps for Naviation, the Bible for Reference, Facebook for Social Networking and Spotify for Music.  

In [65]:
for app in ios_final:
    if app[12] == 'Navigation':
        print(app[2],':',app[6])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Geocaching® : 12811
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5
CoPilot GPS – Car Navigation & Offline Maps : 3582
Google Maps - Navigation & Transit : 154911


In [67]:
for app in ios_final:
    if app[12] == 'Reference':
        print(app[2],':',app[6])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
Merriam-Webster Dictionary : 16849
Google Translate : 26786
Night Sky : 12122
WWDC : 762
Jishokun-Japanese English Dictionary & Translator : 0
教えて!goo : 0
VPN Express : 14
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Real Bike Traffic Rider Virtual Reality Glasses : 8


In [68]:
for app in ios_final:
    if app[12] == 'Social Networking':
        print(app[2],':',app[6])

Facebook : 2974676
LinkedIn : 71856
Skype for iPhone : 373519
Tumblr : 334293
Match™ - #1 Dating App. : 60659
WhatsApp Messenger : 287589
TextNow - Unlimited Text + Calls : 164963
Grindr - Gay and same sex guys chat, meet and date : 23201
imo video calls and chat : 18841
Ameba : 269
Weibo : 7265
Badoo - Meet New People, Chat, Socialize. : 34428
Kik : 260965
Qzone : 1649
Fake-A-Location Free ™ : 354
Tango - Free Video Call, Voice and Chat : 75412
MeetMe - Chat and Meet New People : 97072
SimSimi : 23530
Viber Messenger – Text & Call : 164249
Find My Family, Friends & iPhone - Life360 Locator : 43877
Weibo HD : 16772
POF - Best Dating App for Conversations : 52642
GroupMe : 28260
Lobi : 36
WeChat : 34584
ooVoo – Free Video Call, Text and Voice : 177501
Pinterest : 1061624
知乎 : 397
Qzone HD : 458
Skype for iPad : 60163
LINE : 11437
QQ : 9109
LOVOO - Dating Chat : 1985
QQ HD : 5058
Messenger : 351466
eHarmony™ Dating App - Meet Singles : 11124
YouNow: Live Stream Video Chat : 12079
Cougar 

In [69]:
for app in ios_final:
    if app[12] == 'Music':
        print(app[2],':',app[6])

Pandora - Music & Radio : 1126879
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
Deezer - Listen to your Favorite Music & Playlists : 4677
Sonos Controller : 48905
NRJ Radio : 38
radio.de - Der Radioplayer : 64
Spotify Music : 878563
SoundCloud - Music & Audio : 135744
Sing Karaoke Songs Unlimited with StarMaker : 26227
SoundHound Song Search & Music Player : 82602
Ringtones for iPhone & Ringtone Maker : 25403
Coach Guitar - Lessons & Easy Tabs For Beginners : 2416
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Magic Piano by Smule : 131695
QQ音乐HD : 224
The Singing Machine Mobile Karaoke App : 130
Bandsintown Concerts : 30845
PetitLyrics : 0
edjing Mix:DJ turntable to remix and scratch music : 13580
Smule Sing! : 119316
Amazon Music : 106235
AutoRap by Smule : 18202
My Mixtapez Music : 26286
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
Napster - Top Music 

What is recommended is to bundle some less popular app genres with each other.  Examples are:

- Combine a travel app with a weather features like storm tracking, current weather, forecasts, etc.
- Interactive books on food, drink or finance.
- A health & fitness app that spotlights music.
- A utility app that makes a game out of finishing tasks, paying bills, etc.

Since the iOS app store is dominated with for fun apps, it might stand out to create a productivity app that bundles a for-fun feature.  Adding a fun feature may increase the chances that a user will continue to stay in a productivity app longer so we can offer more in-app purchases opportunities.

Next, we will explore apps by genre in the **Google Play** store:

In [71]:
display_table(android_final,5) #no. of Installs

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


This output places installs into groupings.  In other words, we don't know the precise number of installs for an app.  Instead it tells us that if an app has been downloaded over 1,000 times but less than 10,000 will be counted in the 1,000+ count.  

In order to do a comparison with the categories, we will need to convert these groupings into a float.  So, the app that was counted in the 1,000+ group will be defaulted to 1,000 installs.  We will also need to remove the commas and plus signs otherwise it will not be able to do a float conversion.  We will end the loop by calculating the average number of installs for each category.

In [73]:
cat_android = freq_table(android_final,1)

for cat in cat_android:
    total = 0 #the sum of user ratings
    len_category = 0 #the number of apps specific to ea category
    for app in android_final:
        cat_app = app[1]
        if cat_app == cat:
            n_installs = app[5]
            n_installs = n_installs.replace(',','')
            n_installs = n_installs.replace('+','')
            total += float(n_installs)
            len_category += 1
    avg_and_installs = total/len_category
    print(cat,':',avg_and_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

As with the Apple App store, we have a few big game players that dominate groups of largest installs of over 100 or 500 million installs.

In [76]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] =='1,000,000,000+' or app[5]=='5,000,000,000+' or app[5]=='100,000,000+'):
        print(app[0],':',app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
WeChat : 100,000,000+
Yahoo Mail – Stay Organized : 100,000,000+
BBM - Free Calls & Messages : 100,000,000+


These large download apps like "WhatsApp", "Yahoo Mail", "Android Messages", or "Firefox Browser" make communications apps seem more popular than they actually are.  But, if we look at the categories that were recommended in our Apple app store analysis, we can find some areas that are also popular in the Android market as well.  One example is the `Books and Reference` category.  In the Apple app store, these came up as Books = 39,758.50 and Reference = 74,942.11.  Drilling down into the Android market we see the big game players are:

In [79]:
for app in android_final:
    if app[1]=='BOOKS_AND_REFERENCE' and (app[5]=='1,000,000,000+' or app[5]=='5,000,000,000+' or app[5]=='100,000,000+'):
        print(app[0],':',app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


We can see even though this is a big download category, there aren't alot of big players dominating the genre.  

# Conclusions

In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

Therefore, out of our intial suggestions, an interactive book on food, drink or finance may be a suitable app to develop that will do well in both markets.  This app could include current authors writing short articles.  And in those articles there could be links to the food, drink or finance mentioned.  As well as a social community where people can interact with other users, community experts or the author themself.  This social aspect plays well into both the communication and social networking that is popular in both markets.  It also gives us plenty of opportunity to offer in-app purchases, whereas we can share profits with vendors, the authors and community experts.